db.go

```diff
@@ -755,6 +755,8 @@ var requestPool = sync.Pool{
 }
 
 func (db *DB) writeToLSM(b *request) error {
+	db.lock.RLock()
+	defer db.lock.RUnlock()
```

Comment:
this lock was introduced because it was needed when HandoverSkiplist was introduced - it might not be necessary now

Reply:
removed the lock.